crypto/rand/drbg_lib.c

```diff
@@ -101,26 +145,34 @@ int RAND_DRBG_set(RAND_DRBG *drbg, int type, unsigned int flags)
     int ret = 1;
 
     if (type == 0 && flags == 0) {
-        type = rand_drbg_type;
-        flags = rand_drbg_flags;
+        type = rand_drbg_type[RAND_DRBG_TYPE_MASTER];
+        flags = rand_drbg_flags[RAND_DRBG_TYPE_MASTER];
+    }
+
+    /* If set is called multiple times - clear the old one */
+    if (type != drbg->type && drbg->type != 0 && drbg->meth != NULL) {
+        drbg->meth->uninstantiate(drbg);
     }
 
     drbg->state = DRBG_UNINITIALISED;
     drbg->flags = flags;
     drbg->type = type;
 
-    switch (type) {
-    default:
-        RANDerr(RAND_F_RAND_DRBG_SET, RAND_R_UNSUPPORTED_DRBG_TYPE);
-        return 0;
-    case 0:
+    if (type == 0) {
         /* Uninitialized; that's okay. */
         return 1;
-    case NID_aes_128_ctr:
-    case NID_aes_192_ctr:
-    case NID_aes_256_ctr:
+    }
+    else if (is_ctr(type)) {
         ret = drbg_ctr_init(drbg);
-        break;
+    }
+    else if (is_digest(type)) {
+        if (flags & RAND_DRBG_FLAG_HMAC)
+            ret = drbg_hmac_init(drbg);
+        else
+            ret = drbg_hash_init(drbg);
+    } else {
+        RANDerr(RAND_F_RAND_DRBG_SET, RAND_R_UNSUPPORTED_DRBG_TYPE);
```

Comment:
Here we get a `RANDerr` call but nowhere else!

Reply:
Yes - that's my point. It seems inconsistent that there are no RANDerr calls elsewhere, but there is here. Not sure what the answer is - just pointing it out.